### PR TITLE
Log an error and keep old logs when saving a log larger than the maximum storage capacity

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

* Add missed check in `deleteOldestLogsWithCount` (already fixed in `feature/persistence-priority-critical`, can be discarded on merge)
* Do not delete old logs on try to add a log larger than the maximum storage capacity.

[Update 11/13/2018]:
We re-opened this PR.

### CODE REVIEW

**Comment by jwallra** [comment]
@jaelim-ms @clpolet Do you remember what the reason was that we didn't implement this check in the first place?  I remember whatever implementation we came up with involved checking the size of logs in the db to see if we could fit the new one, which was very expensive.

I think this check looks good though and it prevents customers from shooting themselves in the foot.

**Comment by jaelim-ms** [comment]
@jwallra I remember this is by design.
@MatkovIvan I don't know where a tracking work item for this change. Is it something we've discussed offline or separately?

**Comment by MatkovIvan** [comment]
Discussing offline, closing for now

**Comment by guperrot** [comment]
Waiting after release to restore the changelog change.

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 1]
```diff
@@ -67,6 +67,17 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId flags:(MSFlags)fl
                                    targetKey ? [NSString stringWithFormat:@"'%@'", targetKey] : @"NULL", (unsigned int)persistenceFlags];
   }
   return [self executeQueryUsingBlock:^int(void *db) {
+           // Check maximum size.
+           NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA max_page_count;" inOpenedDatabase:db];
+           NSUInteger maxSize = [(NSNumber *)rows[0][0] intValue] * kMSDefaultDatabaseSizeInBytes;
```
Comment (dhei) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: `kMSDefaultDatabaseSizeInBytes` is default database size, not page size. We need to find a way to use the `kMSDefaultPageSizeInBytes` which was removed by https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1209.

Comment (dhei) [reply AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: we can use `self.page` here

Comment (dhei) [reply AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: addressed my own feedback by https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1204/commits/9f00eaecc3b9dceaf12a596d25e5e2207158e844

File: AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m [hunk 2]
```diff
@@ -67,6 +67,17 @@ - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId flags:(MSFlags)fl
                                    targetKey ? [NSString stringWithFormat:@"'%@'", targetKey] : @"NULL", (unsigned int)persistenceFlags];
   }
   return [self executeQueryUsingBlock:^int(void *db) {
+           // Check maximum size.
+           NSArray<NSArray *> *rows = [MSDBStorage executeSelectionQuery:@"PRAGMA max_page_count;" inOpenedDatabase:db];
```
Comment (clpolet) [inline AppCenter/AppCenter/Internals/Storage/MSLogDBStorage.m]: You can probably use `[MSDBStorage  getMaxPageCountInOpenedDatabase]`. (Which should not have "get" in its name, we missed that)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for iOS and macOS Change Log
 
+## Version 1.12.0 (Under active development)
+
+### AppCenter
+
+* **[Fix]** Do not delete old logs on try to add a log larger than the maximum storage capacity.
```
Comment (clpolet) [inline CHANGELOG.md]: "when trying" instead of "on try" ?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
 # App Center SDK for iOS and macOS Change Log
 
+## Version 1.12.0 (Under active development)
```
Comment (clpolet) [inline CHANGELOG.md]: Should be 1.11.1 I think.

**Pull request merged**